input/blog/2017-09-13-cake-v0.22.0-released.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Cake v0.22.0 released
+category: Release Notes
+author: mholo65
+---
+
+Version 0.22.0 of Cake has been released.
+
+So, let's be clear, this release will break things - but it is all intentional!
+
+This release was initially about updating Roslyn, unifying scripting engines, and targeting .NET 4.6. But one thing led to another which resulted in better support for tool and addin versioning when using preprocessor directives, and an optional in-process NuGet client.
+
+Updating Roslyn and unifying scripting engines has been a long-awaited improvement for many of you. This brings you C# 7 features (without need for `--experimental` switch), debugging with Mono (as we completely removed the old Mono scripting engine), and much, much more. It also allows us to ship Roslyn with Cake, so there's no need for Cake to actually download the Roslyn anymore. In order for us to update Roslyn, we also needed to change target framework to .NET 4.6.
+
+Tool and addin versioning when using preprocessor directives has been a issue which has caused many question. E.g. you may have noted that when changing version for a tool or addin, the old version was still used. Prior to this release, the only solution was to delete the tool/addin from the `tools` folder. This is now fixed as Cake supports side-by-side downloading of different versions of tools and addins. 
+
+Due to unpredictable results when downloading packages using `nuget.exe` tool/addin versioning was solved by adding a intermediate directory containing the version number. This intermediate directory can however cause `MAX_PATH` issues when downloading some specific packages. To address this issue, we implemented functionality in `Cake.NuGet` to allow Cake to download tools and addins without the need for `nuget.exe`. This functionality is currently optional through [configuration](../docs/fundamentals/configuration.md), and is the first step towards removing need for `nuget.exe` in Cake. Example usage to opt-in for the in-process NuGet client:
+```
+Cake.exe --NuGet_UseInProcessClient=true
```

Comment:
Could we change the settings to `snake_case` here?

Reply:
I kind of like PascalCase :wink:
having that said, I'm all for democracy, so let's see what others think.

Reply:
It's an URI gosh darn it! 😉

Reply:
Environment variables upper case, args lower case, configuration files pascal case.

Reply:
The existing documentation uses `snake_case`: https://cakebuild.net/docs/fundamentals/default-configuration-values